signal/micro/kernels/xtensa/fft_auto_scale.cc

```diff
@@ -0,0 +1,101 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "signal/src/fft_auto_scale.h"
+
+#include <math.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "signal/micro/kernels/fft_auto_scale.h"
+#include "signal/src/max_abs.h"
+#include "signal/src/msb.h"
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/micro_context.h"
+
+#if XCHAL_HAVE_HIFI3
+#include <xtensa/tie/xt_hifi3.h>
+// Implementation for DSPs that support the Hifi3 ISA. Bit exact with the
+// portable version below.
+int FftAutoScale(const int16_t* input, int size, int16_t* output) {
```

Comment:
We should still put this in an anonymous namespace. I would suggest using the same one as Eval below, but name this XtensaFftAutoScale to ensure no lookup issues.

Reply:
Done.